shared/Utils.js

```diff
@@ -3,5 +3,50 @@
 export default {
 	stripHtmlTags (inputString) {
 		return inputString.replace(/<\/?[^>]+(>|$)/g, "");
+	},
+
+	/**
+	 * Generate the navigation menu based on the response from the Wordpress API.
+	 * The retun menu array will:
+	 * 1. Exclude all pages whose menu_order === 0;
+	 * 2. Append "News" and "Views" menu items to the end.
+	 * @param {Array<Object>} sitePages - All menu pages sorted by menu_order.
+	 * @param {Boolean} includeHomeItem - A flag indicating if the home menu item should be included.
+	 * @return {Array<Object>} An array of object including the slug, menu title and href.
+	 */
+	generateNavMenu (sitePages, includeHomeItem) {
+		// Don't include the menu item for "home" by default
+		includeHomeItem = includeHomeItem === undefined ? false : includeHomeItem;
+		const menuItems = [];
+		const homeMenuItem = [];
+		const itemsToAppend = [{
+			slug: "news",
+			title: "News",
+			href: "/news"
+		}, {
+			slug: "views",
+			title: "Views",
+			href: "/views"
+		}];
+
+		sitePages.forEach((oneSitePage) => {
+			if (includeHomeItem && oneSitePage.slug === "home") {
+				homeMenuItem.push({
+					slug: oneSitePage.slug,
+					title: "Home",
+					href: "/"
+				});
+			}
+
+			if (oneSitePage.menu_order > 0) {
```

Comment:
what does `oneSitePage.menu_order` contain? I'm not really sure why it would ever contain a negative value. And it doesn't look like you're using the order to sort the menu items.

Reply:
`oneSitePage.menu_order` could be 0 or non-zero. I will change the condition to `oneSitePage.menu_order !== 0` and add a code comment to reduce the confusion.

The sort is performed via [the fetch from WP API](https://github.com/cindyli/wecount.inclusivedesign.ca/blob/auto-generate-nav-menu/shared/DataFetcher.js#L61) with `order=asc&orderby=menu_order`.

Reply:
@cindyli, I think you could write `if(oneSitePage.menu_order)` or maybe I'm mistaking what you're saying.

Reply:
Sure, I will change it to `if (oneSitePage.menu_order)` although my personal preference is to be explicit about what exact condition `if` should handle.

Reply:
Ok, that's reasonable for keeping it as you suggested.